discovery/sync_manager.go

```diff
@@ -322,7 +323,9 @@ func (m *SyncManager) syncerHandler() {
 			// keep track of the corresponding syncer to properly
 			// handle disconnects. We'll also use a signal to know
 			// when the historical sync completed.
-			setInitialHistoricalSyncer(s)
+			if !isPinnedSyncer {
```

Comment:
hm, why this condition?

Reply:
Several reason, also mentioned in another comment w/ wilmer. 

Each pinned peer already does a historical sync, so there can be many of them at the same time. When it comes to replacement, ideally you'd prioritize pinned over passive syncers but the bookkeeping gets tricky when they're all happening in parallel. Disconnects need to be handled differently depending on pinned vs regular. Tl;dr the number of edge cases isn't worth it, and it's simpler to reason about them as separate peer groups.

My preferred approach if we are interested in the cosmetic aspect is just a callback that syncers can use to report they've completed a historical sync.

Reply:
Yeah, I'm kinda worried that all these different states can become hard to reason about and maintain. Perhaps not in the scope of this PR, but it could use an overhaul